src/libcollections/str.rs

```diff
@@ -1683,6 +1684,32 @@ impl str {
     pub fn parse<F: FromStr>(&self) -> Result<F, F::Err> {
         core_str::StrExt::parse(self)
     }
+    
+    /// Returns a string slice extracted from the starting index to
+    /// the specified length.
+    ///
+    /// Returns [`None`] if length is zero, or if string is empty.
+    /// # Example
```

Comment:
Doc needs update for change of functionality. You could compare it with usual slicing (`&s[a..b]` style).